addons/website_forum/views/website_forum.xml

```diff
@@ -360,7 +412,15 @@
 <!-- Edition: Post Article -->
 <template id="new_link">
     <t t-call="website_forum.header">
-        <h1 class="mt0">Submit a Link</h1>
+        <t t-set="link_create" t-value="False if post_name else True"/>
```

Comment:
Before this it is not possible to edit previously posted Link.
But now with this you can make change in previously posted links. 
I'm using same create form for edit links. Just changing action_url and string to 'Edit your Link' from 'Submit a Link' This `link_create` is holding a state weather this is edit or create